docs/investigations/rest-adapters/rest-adapter-for-graphql.md

```diff
@@ -0,0 +1,103 @@
+not introducing previously mentioned controller, and just instruct Admins to create application using Compass UI. 
+
+2. Get the client certificate
+
+When creating Token Request, a  Controller updates the status and provide token and configuration URL fields.
+Controller needs to be reconfigured, to point to the REST Adapter endpoint `/v1/applications/signingRequests/info`. Token is generated via sending request to `/v1/applications/tokens` and this also has to be reconfigured to point to the Adapter Endpoint.
+REST Adapter forwards request to mutation `requestOneTimeTokenForApplication`.
+This mutation consumes applicationID, so we need to have mapping from application name to application ID in Compass (to check if it is stored in CR).
+
+Calling `/v1/applications/signingRequests/info` endpoint, following data is returned:
+```json
+
+{
+    "csrUrl": "",
+    "api":{
+        "metadataUrl":      "",
+        "eventsUrl":        "",
+        "infoUrl":          "",
+        "certificatesUrl":  "",
+    },
+    "certificate":{
+        "subject":"OU=Test,O=TestOrg,L=Waldorf,ST=Waldorf,C=DE,CN={APP_NAME}",
+        "extensions": "",
+        "key-algorithm": "rsa2048",
+    }
+}
+```
+
+`csrUrl` should point to the adapter's `/v1/applications/certificates`.
+`certificate` field is populated by adapter by calling Connector's GraphQL query `configuration`. 
+
+3. Register a service
+
+```
+curl -k --cert ./app1_client.crt   --key ./generated.key -X POST https://gateway.34.77.12.120.xip.io/app1/v1/metadata/services --data-binary "@serviceRegistrationManyAPIs.json"
+```
+```json
+{
+  "provider":"SAP",
+  "name":"api1",
+  "description": "api1 desc",
+  "shortDescription": "api1 short desc",
+  "identifier": "api1Identifier",
+  "labels":{
+  	"label1":"value1",
+  	"label2": "value2"
+  },
+  "api":{
+  	"targetUrl":"http://some-app1.pl",
+  	"requestParameters": {
+  	  "headers": {
+  	      "header1": "value1"
+  	  }
+  	}
+  }
+
+}
+```
+Compass equivalent: `addApiDefinition` or `addEventDefinition`
+
+Challenges:
+
+- in old API, there is a Service which seems to be equivalent to API Package, which is missing in Compass
+- `identifier` has no mapping in Compass API
+- you cannot label service class via Compass API
+- specificationURL, specificationCredentials, specificationRequestParameters can be mapped to FetchRequest, but it is not yet implemented
+- from Documentation:
+> If the api.spec or api.specificationUrl parameters are not specified and the api.type parameter is set to OData, the Application Registry will try to fetch the specification from the target URL with the $metadata path.
+
+Special handling for OData can be implemented in the Adapter (assuming that FetchRequest is implemented)
+
+TODO: CSRF tokens are not implemented in Compass
+- documentation: in Compass, documentation can be added only on the Application level
+
+# Diagram
```

Comment:
The adapters are not a part of the Kyma Runtime. They are components that are run inside the Management Plane. I would put them in front of the GraphQL API components.

If we go with the TokenRequest CR then it is a part of the Kyma Runtime.